lib/elements/rebar.rb

```diff
@@ -0,0 +1,66 @@
+require './db/base'
+require './lib/materials/steel'
+
+module Elements
+  class Rebar < Materials::Steel
+    MAX_YIELD_STRESS = 550.to_f
+
+    attr_reader :diameter, :yield_stress, :number, :hooks_class, :elastic_module, :start_hook, :end_hook
+
+    def self.build_by_standard(number:, yield_stress:)
+      rebar_data = DB::Base::STANDARD_REBAR.find { |bar_data| bar_data['number'] == number }
+
+      @number = number
+
+      new(
+        yield_stress:,
+        diameter: rebar_data['diameter'],
+        elastic_module: nil
+      )
+    end
+
+    def initialize(yield_stress:, elastic_module:, diameter:)
+      @diameter = diameter.to_f
+      @start_hook = nil
+      @end_hook = nil
+
+      super(yield_stress:, elastic_module:)
+    end
+
+    def design_yield_stress
+      [yield_stress, MAX_YIELD_STRESS].min
+    end
```

Comment:
is this a constraint from design codes?

Reply:
Yes, but taking a look into my notes, I think is a construction constraint.